app/controllers/discussions_controller.rb

```diff
@@ -77,9 +78,19 @@ def unfavorite
     redirect_to discussions_path(discussion_id: params[:discussion_id], search: params[:search])
   end
 
+  def upvote
+    if current_user
```

Comment:
Also, currently if you're not signed in, you can still see the "^" button that allows you to vote and it shows that it changes the score from 0 to 1, but when you refresh the page it doesn't actually save it (since you're not signed in as a user).
We shouldn't allow users to even try to vote when they're not signed in - so let's not show the "^" unless a user is signed in